src/pollManager.ts

```diff
@@ -2,34 +2,38 @@ import { Logger } from '@map-colonies/js-logger';
 import { inject, singleton } from 'tsyringe';
 import { Watcher } from './watcher';
 import { Readiness } from './probe/readiness';
-import { Liveness } from './probe/liveness';
 import { Services } from './common/constants';
-import { IPollConfig } from './common/interfaces';
+import { IConfigProvider, IPollConfig } from './common/interfaces';
 
 @singleton()
 export class PollManager {
   public constructor(
     @inject(Services.LOGGER) private readonly logger: Logger,
     @inject(Services.POLLCONFIG) private readonly pollCofig: IPollConfig,
+    @inject(Services.CONFIGPROVIDER) private readonly configProvider: IConfigProvider,
     private readonly watcher: Watcher,
-    private readonly liveness: Liveness,
     private readonly readiness: Readiness
   ) {}
 
   public async poll(): Promise<void> {
     const frequencyTimeOutMS = this.pollCofig.timeout.frequencyMilliseconds;
-    const livenessKillTimeOutSeconds = this.pollCofig.timeout.livenessKillSeconds;
+    const requestsKillSeconds = this.pollCofig.timeout.requestsKillSeconds;
+    const afterUpdateDelaySeconds = 0.1;
     try {
       this.logger.info(`polling attempt`);
       if (!(await this.watcher.isUpdated())) {
-        this.logger.debug('changes detected! - updating configurations');
+        this.logger.debug('changes detected!');
         const readinessKillRndInt = this.getRandomInteger();
         this.logger.info(`killing readiness in ${readinessKillRndInt} seconds`);
         await this.delay(readinessKillRndInt);
         this.readiness.kill();
-        this.logger.info(`killing liveness in ${livenessKillTimeOutSeconds} seconds`);
-        await this.delay(livenessKillTimeOutSeconds);
-        this.liveness.kill();
+        this.logger.info(`killing existing requests in ${requestsKillSeconds}`);
+        await this.delay(requestsKillSeconds);
+        this.logger.info('updating configurations');
+        await this.configProvider.createOrUpdateConfigFile();
+        // make sure mapproxy start updating before restart readiness
+        await this.delay(afterUpdateDelaySeconds);
```

Comment:
i think this should come from configuration - it will take time for the mapproxy to load after config change

Reply:
fixed in 1b5052a